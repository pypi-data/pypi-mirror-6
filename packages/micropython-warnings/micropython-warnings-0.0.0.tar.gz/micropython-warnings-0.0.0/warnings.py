# MicroPython doesn't support warnings
