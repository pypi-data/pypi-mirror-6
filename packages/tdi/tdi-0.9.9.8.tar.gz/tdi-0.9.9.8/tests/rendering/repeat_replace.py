#!/usr/bin/env python
import warnings as _warnings
_warnings.resetwarnings()
_warnings.filterwarnings('error')

from tdi import html

template = html.from_string("""
<node tdi="item"><xnode tdi="xitem"><ynode tdi="yitem">
<a tdi="-a" />
</ynode></xnode></node>
""".lstrip())

class Model(object):
    def render_item(self, node):
        node.xitem.yitem
        node.repeat(None, [1, 2, 3, 4])

    def render_xitem(self, node):
        ctx = node.ctx
        node.replace(None, node.yitem).ctx = ctx

    def render_a(self, node):
        node.content = node.ctx[1]

model = Model()
template.render(model)
