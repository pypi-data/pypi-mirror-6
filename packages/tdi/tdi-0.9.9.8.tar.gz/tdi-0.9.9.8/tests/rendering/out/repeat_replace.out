<node><ynode>
1
</ynode></node><node><ynode>
2
</ynode></node><node><ynode>
3
</ynode></node><node><ynode>
4
</ynode></node>
